src/components/learner-credit-management/MultipleBudgetsPicker.jsx

```diff
@@ -16,27 +19,76 @@ const MultipleBudgetsPicker = ({
   enableLearnerPortal,
 }) => {
   const orderedBudgets = orderBudgets(budgets);
+
+  const rows = useMemo(
+    () => orderedBudgets.map(budget => {
+      const budgetLabel = getBudgetStatus({
+        startDateStr: budget.start,
+        endDateStr: budget.end,
+        isBudgetRetired: budget.isRetired,
+      });
+
+      return ({
+        ...budget,
+        status: budgetLabel.status,
+        enterpriseUUID,
+        enterpriseSlug,
+        enableLearnerPortal,
+      });
+    }),
+    [orderedBudgets, enterpriseUUID, enterpriseSlug, enableLearnerPortal],
+  );
+
+  const reducedChoices = orderedBudgets.reduce((acc, currentObject) => {
+    const budgetLabel = getBudgetStatus({
+      startDateStr: currentObject.start,
+      endDateStr: currentObject.end,
+      isBudgetRetired: currentObject.isRetired,
+    });
+
+    if (budgetLabel.status in acc) {
+      acc[budgetLabel.status].number += 1;
+    } else {
+      acc[budgetLabel.status] = {
+        name: budgetLabel.status,
+        number: 1,
+        value: budgetLabel.status,
+      };
+    }
+    return acc;
+  }, {});
```

Comment:
nit: Can reduce the complexity a bit with [lodash.groupBy](https://lodash.com/docs/4.17.15#groupBy)
```suggestion
    const budgetLabels = orderedBudgets.map( budget => {
         startDateStr: budget.start,
        endDateStr: budget.end,
        isBudgetRetired: budget.isRetired,
     });
    const budgetLabelsByStatus = groupBy(budgetLabels, 'status');
    const reducedChoices = Object.keys(budgetLabelsByStatus).map( budgetLabel => {
        name: budgetLabel,
        number: budgetLabelsByStatus[budgetLabel].length,
        value: budgetLabel,
      });
```